watertap/unit_models/anaerobic_digester.py

```diff
@@ -957,11 +968,11 @@ def initialize_build(
             optarg = {}
 
         # Check DOF
-        if degrees_of_freedom(self) != 0:
-            raise InitializationError(
-                f"{self.name} degrees of freedom were not 0 at the beginning "
-                f"of initialization. DoF = {degrees_of_freedom(self)}"
-            )
+        # if degrees_of_freedom(self) != 0:
+        #     raise InitializationError(
+        #         f"{self.name} degrees of freedom were not 0 at the beginning "
+        #         f"of initialization. DoF = {degrees_of_freedom(self)}"
+        #     )
```

Comment:
Why is this line being commented out?

Reply:
This one is commented out for passing the initialization, if we include this check, then after `propagate_state`, they still raise the error. Maybe @agarciadiego has an idea on how to rewrite this check

Reply:
Looking at other unit models, it seems like this DOF check is before the initialization in some cases and after the initialization in others (most unit models actually don't include this check at all). I'm not sure why one option should be chosen over the other, but if we move this check to after the initialization, it seems like everything passes on my end.

Reply:
It's prob best if this commented code is moved to the end of the initialization routine or just deleted altogether

Reply:
Given this is an initialization routine, it is important to check that the DoF are 0 before starting (or at least what you expect them to be). If you try initializing a model with degrees of freedom, then how do you know that the answer will actually be meaningful?

If this check is failing, it indicates that something is going on that you really need to investigate.

You should not be checking this after initialization as part of the routine - you definitely should have a **test** that checks for that however.

Finally, I am not sure on WaterTAPs plans for this, but note that IDAES has a new initialization API that handles a lot of this for you and that we do plan to eventually deprecate the current API.

Reply:
@luohezhiming I'm not sure why the initialization routine in `modified_ADM1_flowsheet` wasn't working as intended, but I've updated it to use SequentialDecomposition, and now the DOF check works fine.